docs/manual/developer_guide.adoc

```diff
@@ -1337,34 +1337,421 @@ Available low-level Jinja macros that can be used in Bash remediations:
+** *daemonname* - name of the daemon. This argument is optional. If *daemonname* is not specified it means the name of the daemon is the same as the name of service.
+** *mask_service* - Check if the service should be masked to prevent it from running. This argument is optional. Implicit value is "true".
+* Languages: Ansible, Bash, OVAL, Puppet
+
+service_enabled::
+* Checks if a system service is enabled. Uses either systemd or SysV init based on the product configuration in `product.yml`.
+* Parameters:
+** *servicename* - name of the service.
+** *packagename* - name of the package that provides this service. This argument is optional. If *packagename* is not specified it means the name of the package is the same as the name of service.
+** *daemonname* - name of the daemon. This argument is optional. If *daemonname* is not specified it means the name of the daemon is the same as the name of service.
+* Languages: Ansible, Bash, OVAL, Puppet
+
+sshd_lineinfile::
+* Checks SSH server configuration items in `/etc/ssh/sshd_config`.
+* Parameters:
+** *parameter* - name of the SSH configuration option, eg. `KerberosAuthentication`
+** *value* - value of the SSH configuration option specified by *parameter*, eg. `"no"`.
+** *missing_parameter_pass* - If set to `"true"` the OVAL check will pass if the parameter is not present in `/etc/ssh/sshd_config`.
+* Languages: Ansible, Bash, OVAL
+
+sysctl::
+* Checks sysctl parameters. The OVAL definition checks both configuration and runtime settings and require both of them to be set to the desired value to return true.
+* Parameters:
+** *sysctlvar* - name of the sysctl value, eg. `net.ipv4.conf.all.secure_redirects`.
+** *datatype* - data type of the sysctl value, eg. `int`.
+** *sysctlval* - value of the sysctl value, eg. `'1'`. If this parameter is not specified, XCCDF Value is used instead.
+* Languages: Ansible, Bash, OVAL
+
+timer_enabled::
+* Checks if a SystemD timer unit is enabled.
+* Parameters:
+** *timername* - name of the SystemD timer unit, without the `timer` suffix, eg. `dnf-automatic`.
+** *packagename* - name of the RPM package which provides the SystemD timer unit. This parameter is optional, if it is not provided it is assumed that the name of the RPM package is the same as the name of the SystemD timer unit.
+* Languages: Ansible, Bash, OVAL
+
+
+==== Creating Templates
+
+The offer of currently available templates can be extended by developing a new
+template.
+
+1) Create the template files, one for each type of file. Each one should be
+named `template_<TYPE>_<NAME>`. Where `<TYPE>` should be OVAL, ANSIBLE, BASH,
+ANACONDA or PUPPET and `<NAME>` is the what we will call the template name.
+Create these files in `link:{rootdir}/shared/templates[shared/templates]` directory.
+
+Use the Jinja syntax we use elsewhere in the project; refer to the earlier
+section on Jinja macros for more information.  The parameters should be named
+using lowercase letters.
```

Comment:
What do you mean by named using lower case letters? I am confused because in the example oval template the parameters are in upper case.

Reply:
Should be uppercase there at this moment. I'll fix that. But I think using lowercase letters this is the desired future state, because using uppercase letters is only a temporary workaround for easier transition from old system to the new system.

https://github.com/ComplianceAsCode/content/blob/85bdd8bbb8753f77f6c728811c89f026b76b4d8b/ssg/templates.py#L318-L323